### PR TITLE
Validate options in CREATE FOREIGN TABLE

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: server/src/main/java/io/crate/fdw/JdbcForeignDataWrapper.java [hunk 1]
```diff
@@ -58,6 +58,14 @@ final class JdbcForeignDataWrapper implements ForeignDataWrapper {
     private final Setting<String> urlSetting = Setting.simpleString("url");
     private final List<Setting<?>> mandatoryServerOptions = List.of(urlSetting);
 
+    private final Setting<String> schemaName = Setting.simpleString("schema_name");
+    private final Setting<String> tableName = Setting.simpleString("table_name");
+    private final List<Setting<?>> optionalTableOptions = List.of(
```
[the remark this answers is not in the document]
Comment (mfussenegger) [reply server/src/main/java/io/crate/fdw/JdbcForeignDataWrapper.java:63]: No:

```
:schema_name:

  The schema name used when accessing a table in the foreign system. If not
  specified this defaults to the schema name of the table created within
  CrateDB.

  Use this if the names between CrateDB and the foreign system are different.

:table_name:

  The table name used when accessing a table in the foreign system. If not
  specified this defaults to the table name of the table within CrateDB.

  Use this if the names between CrateDB and the foreign system are different.
```

[the remark this answers is not in the document]
Comment (matriv) [reply server/src/main/java/io/crate/fdw/JdbcForeignDataWrapper.java:63]: ah, thank you!

**Review by matriv — APPROVED** [review]
LGTM, i have this other, optional, comment to improve the error msg test.

File: server/src/test/java/io/crate/fdw/ForeignDataWrapperPlannerTest.java [hunk 2]
```diff
@@ -110,4 +113,35 @@ public void test_cannot_create_server_if_fdw_is_missing() throws Exception {
...
+            "pg",
+            "jdbc",
+            "crate",
+            true,
+            Settings.builder().put("url", "jdbc:postgresql://localhost:5432/").build()
+        );
+        AddServerTask addServerTask = new AddServerTask(e.foreignDataWrappers, request);
+        ClusterServiceUtils.setState(clusterService, addServerTask.execute(clusterService.state()));
+
+        String stmt = "create foreign table tbl (x int) server pg options (invalid 42)";
```
[the remark this answers is not in the document]
Comment (mfussenegger) [reply server/src/test/java/io/crate/fdw/ForeignDataWrapperPlannerTest.java:130]: I think the option order is currently not stable (hashmap) and it's already going through the `Collectors.joining(", ")` to print the single item so I think coverage is good enough as we can rely on `Collectors.joining` working.

**Pull request merged**